src/main/groovy/com/cinnober/gradle/semver_git/SemverGitPlugin.groovy

```diff
@@ -24,8 +24,110 @@
+import org.gradle.api.Project
+import org.gradle.api.provider.Property
+
+interface SemverGitPluginExtension {
+    /**
+     * One of <code>"minor"</code> (default) or <code>"major"</code> to bump the respective SemVer part,
+     * or a literal value such as <code>"3.0.0-rc2"</code> to use as the next version.
+     */
+    Property<String> getNextVersion()
+
+    /**
+     * Pattern for the suffix to append when the current commit is not version tagged.
+     *
+     * <p>
+     * Example: <code>"&lt;count&gt;.g&lt;sha&gt;&lt;dirty&gt;-SNAPSHOT"</code>.
+     * </p>
+     *
+     * <p>
+     * Pattern may include the following placeholders:
+     * </p>
+     *
+     * <ul>
+     *   <li><code>&lt;count&gt;</code>: Number of commits (including this) since the last version tag.</li>
+     *   <li><code>&lt;sha&gt;</code>: Abbreviated ID of the current commit.</li>
+     *   <li><code>&lt;dirty&gt;</code>: The value of {@link #getDirtyMarker() dirtyMarker} if there are unstaged changes,
+     *   otherwise empty. Note that untracked files do not count.</li>
+     * </ul>
+     *
+     * <p>
+     * Default: <code>"SNAPSHOT"</code>
+     * </p>
+     */
+    Property<String> getSnapshotSuffix()
+
+    /**
+     * The value to substitute for <code>&lt;dirty&gt;</code> in {@link #getSnapshotSuffix() snapshotSuffix}
+     * if there are unstaged changes.
+     *
+     * <p>
+     * Note that this has no effect if {@link #getSnapshotSuffix() snapshotSuffix}
+     * does not include a <code>&lt;dirty&gt;</code> placeholder.
+     * </p>
+     *
+     * <p>
+     * Default: <code>"-dirty"</code>
+     * </p>
+     */
+    Property<String> getDirtyMarker()
+
+    /**
+     * Additional arguments for the <code>git describe</code> subprocess.
+     *
+     * <p>
+     * Default: <code>"--match *[0-9].[0-9]*.[0-9]*"</code>
+     * </p>
+     */
+    Property<String> getGitDescribeArgs()
+
+    /**
+     * A prefix that may be stripped from tag names to create a version number.
+     *
+     * <p>
+     * For example: if your tags are named like <code>v3.0.0</code>, set this to <code>"v"</code>.
+     * </p>
+     *
+     * <p>
+     * Default: null
+     * </p>
+     */
+    Property<String> getPrefix()
+}
+
+class DeferredVersion {
+    private final Project project;
+    private final SemverGitPluginExtension extension;
+    private String version = null;
+
+    DeferredVersion(Project project, SemverGitPluginExtension extension) {
+        this.project = project;
+        this.extension = extension;
+    }
+
+    private synchronized String evaluate() {
+        if (version == null) {
+            version = SemverGitPlugin.getGitVersion(
+              this.extension.nextVersion.getOrNull(),
+              this.extension.snapshotSuffix.getOrNull(),
+              this.extension.dirtyMarker.getOrNull(),
+              this.extension.gitDescribeArgs.getOrNull(),
+              this.extension.prefix.getOrNull(),
+              this.project.projectDir
```

Comment:
Since the `Project` is only used for `projectDir` can we remove project entirely?
Makes it a little easier to reason about and also doesn't prevent additional blocks for configuration-cache compatibility

Reply:
Sure!